testsuite/fuzzer/fuzz/fuzz_targets/move/utils.rs

```diff
@@ -1,8 +1,264 @@
+// used for ordering modules topologically
+pub(crate) fn sort_by_deps(
+    map: &BTreeMap<ModuleId, CompiledModule>,
+    order: &mut Vec<ModuleId>,
+    id: ModuleId,
+    visited: &mut HashSet<ModuleId>,
+) -> Result<(), Corpus> {
+    if visited.contains(&id) {
+        return Err(Corpus::Keep);
+    }
+    visited.insert(id.clone());
+    if order.contains(&id) {
+        return Ok(());
+    }
+    let compiled = &map.get(&id).unwrap();
+    for dep in compiled.immediate_dependencies() {
+        // Only consider deps which are actually in this package. Deps for outside
+        // packages are considered fine because of package deployment order. Note
+        // that because of this detail, we can't use existing topsort from Move utils.
+        if map.contains_key(&dep) {
+            sort_by_deps(map, order, dep, visited)?;
+        }
+    }
+    order.push(id);
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn publish_transaction_payload(modules: &[CompiledModule]) -> TransactionPayload {
+    let modules_metadatas: Vec<_> = modules
+        .iter()
+        .map(|cm| ModuleMetadata {
+            name: cm.name().to_string(),
+            source: vec![],
+            source_map: vec![],
+            extension: MoveOption::default(),
+        })
+        .collect();
+
+    let all_immediate_deps: Vec<_> = modules
+        .iter()
+        .flat_map(|cm| cm.immediate_dependencies())
+        .map(|mi| PackageDep {
+            account: mi.address,
+            package_name: mi.name.to_string(),
+        })
+        .collect::<BTreeSet<_>>() // leave only uniques
+        .into_iter()
+        .filter(|c| &c.account != modules[0].address()) // filter out package itself
+        .collect::<Vec<_>>();
+
+    let metadata = PackageMetadata {
+        name: "fuzz_package".to_string(),
+        upgrade_policy: UpgradePolicy::compat(), // TODO: currently does not matter. Maybe fuzz compat checks specifically at some point.
+        upgrade_number: 1,
+        source_digest: "".to_string(),
+        manifest: vec![],
+        modules: modules_metadatas,
+        deps: all_immediate_deps,
+        extension: MoveOption::default(),
+    };
+    let pkg_metadata = bcs::to_bytes(&metadata).expect("PackageMetadata must serialize");
+    let mut pkg_code: Vec<Vec<u8>> = vec![];
+    for module in modules {
+        let mut module_code: Vec<u8> = vec![];
+        module
+            .serialize(&mut module_code)
+            .expect("Module must serialize");
+        pkg_code.push(module_code);
+    }
+    code_publish_package_txn(pkg_metadata, pkg_code)
+}
+
+#[allow(dead_code)]
+// panic to catch invariant violations
+pub(crate) fn check_for_invariant_violation(e: VMStatus) {
+    if e.status_type() == StatusType::InvariantViolation {
+        // known false positive
+        if e.message() != Some(&"moving container with dangling references".to_string()) {
+            panic!("invariant violation {:?}", e);
+        }
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
nit: here and in other places, these things are not dead code?

Reply:
Yes, we marked everything to be ignored by clippy in that file. I can probably move the exclusion on file level, in case we stop using any of these functions, but most likely we want to keep them around without complaints from clippy.